addons/sms/tools/sms_tools.py

```diff
@@ -0,0 +1,13 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
+
+import markupsafe
+
+from odoo.tools import html_escape, html_keep_url
+
+
+def sms_content_to_rendered_html(text):
+    """Transforms plaintext into html making urls clickable and preserving newlines"""
+    text_with_links = html_keep_url(str(html_escape(text)))
+    return markupsafe.Markup(re.sub(r'\r?\n|\r', '<br/>', text_with_links))
```

Comment:
Hello @tde-banana-odoo,

 I've had to add this to avoid the warning passing `body_is_html=True` in `message_post` after we've processed the body with this method.

Note that we've escaped `text` to get `text_with_links` just above.